api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -4543,4 +4543,25 @@ default NDArray norm(int[] axes, boolean keepDims) {
      * @return the norm of this {@code NDArray}
      */
     NDArray norm(int ord, int[] axes, boolean keepDims);
+
+    /**
+     * Returns a one-hot {@code NDArray}.
+     *
+     * @param depth Depth of the one hot dimension.
+     * @param onValue The value assigned to the locations represented by indices.
+     * @param offValue The value assigned to the locations not represented by indices.
+     * @param dataType dataType of the output.
+     * @return one-hot encoding of this {@code NDArray}
+     */
+    NDArray oneHot(int depth, float onValue, float offValue, DataType dataType);
```

Comment:
Is this operator commonly used in the mathmatical operations? Or it is jusr for BERT model?

Reply:
It is often used to convert categorical value tokens to a more expressive feature vector, not just for BERT model. Check [this](https://d2l.djl.ai/chapter_linear-networks/softmax-regression.html#classification-problems) and [this](https://d2l.ai/chapter_recurrent-neural-networks/rnn-scratch.html) for an example.